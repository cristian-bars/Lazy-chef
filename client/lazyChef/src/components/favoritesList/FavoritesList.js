import React from 'react';
import {View, Text, FlatList, Image, TouchableOpacity} from 'react-native';
import {connect} from 'react-redux';
import {PropTypes} from 'prop-types';
import {useNavigation} from '@react-navigation/native';
import styles from './favoritesListStyles';

const RecipesList = ({recipes, dispatch, userAccess}) => {
  const recipesList = recipes;

  const myRecipesIds = userAccess.user.favouriteRecipes;
  let myRecipe = [];
  let myRecipesList = [];
  if (myRecipesIds.length) {
    myRecipesIds.forEach(id => {
      myRecipe = recipesList.find(recipe => recipe._id === id);
      myRecipesList.push(myRecipe);
    });
  }

  console.log(myRecipesList);
  const navigation = useNavigation();

  const listRender = ({item}) => {
    return (
      <TouchableOpacity
        style={styles.recipeDetail}
        onPress={() => navigation.navigate('Detail', {recipeId: item._id})}>
        {item.image[0] ? (
          <Image style={styles.logo} source={{uri: item.image[0]}} />
        ) : (
          <Image
            style={styles.logo}
            source={{
              uri: 'https://i.ibb.co/PwFdwdH/Vector-artistic-pen-and-ink-drawing-illustration-of-empty-plate-knife-and-fork.jpg',
            }}
          />
        )}

        <Text style={styles.titleText}>{item.title}</Text>
        <View style={styles.recipeIcons}>
          <View>
            <Image
              style={styles.imageIcons}
              source={require('../../img/Ingredientes.jpg')}
            />
            <Text style={styles.descriptionText}>
              {item.recipeIngredient.length}
            </Text>
          </View>
          <View>
            <Image
              style={styles.imageIcons}
              source={require('../../img/herramientas-y-utensilios.png')}
            />
            <Text style={styles.descriptionText}>{item.difficulty}</Text>
          </View>
          <View>
            <Image
              style={styles.imageIcons}
              source={require('../../img/duration.png')}
            />
            <Text style={styles.descriptionText}>{item.totalTime}</Text>
          </View>
        </View>

        <Text style={styles.descriptionText}>{item.Description}</Text>
      </TouchableOpacity>
    );
  };
  return (
    <View style={styles.container}>
      <View>
        {recipes.length ? (
          <FlatList
            style={styles.list}
            data={myRecipesList}
            keyExtractor={item => item.title}
            renderItem={listRender}
          />
        ) : (
          <Text>No tenim receptes</Text>
        )}
      </View>
    </View>
  );
};

RecipesList.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(store) {
  return {
    recipes: store.recipes,
    userAccess: store.userAcces,
  };
}

export default connect(mapStateToProps)(RecipesList);
