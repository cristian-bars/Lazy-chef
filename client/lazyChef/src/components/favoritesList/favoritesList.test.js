import 'react-native';
import React from 'react';
import FavoritesList from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react-native';

describe('Given a FavoritesList component', () => {
  const mockStore = Store([thunk]);
  let myStore;
  beforeEach(() => {
    myStore = mockStore({
      userAcces: {
        user: {
          email: 'Cristian@gmail.com',
          password: '1a',
          favouriteRecipes: ['4a3b2c1d'],
        },
      },
      recipes: [
        {
          title: 'meat',
          _id: '4a3b2c1d',
          image: ['image1'],
          recipeIngredient: ['perejil', 'tomate'],
        },
        {
          _id: '123456',
          title: 'fish',
          image: ['image2'],
          recipeIngredient: ['agua', 'arroz'],
        },
      ],
    });
  });

  test('renders correctly', () => {
    const wrapper = render(
      <Provider store={myStore}>
        <FavoritesList />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
