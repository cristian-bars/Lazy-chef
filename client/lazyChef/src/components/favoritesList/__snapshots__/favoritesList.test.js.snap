// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a FavoritesList component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#D3E0FF",
    }
  }
>
  <View>
    <RCTScrollView
      data={
        Array [
          Object {
            "_id": "4a3b2c1d",
            "image": Array [
              "image1",
            ],
            "recipeIngredient": Array [
              "perejil",
              "tomate",
            ],
            "title": "meat",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "flexDirection": "column",
                "margin": 10,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "image1",
                }
              }
              style={
                Object {
                  "height": 150,
                }
              }
            />
            <Text
              style={
                Object {
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "padding": 10,
                }
              }
            >
              meat
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                }
              }
            >
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/Ingredientes.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                >
                  2
                </Text>
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/herramientas-y-utensilios.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/duration.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 15,
                  "padding": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
